gapis/validate/validate.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"context"
+	"io/ioutil"
+
+	perfetto_pb "perfetto/config"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/gapid/core/app/crash"
+	"github.com/google/gapid/core/event/task"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/core/os/android/adb"
+	"github.com/google/gapid/gapis/perfetto"
+	perfetto_android "github.com/google/gapid/gapis/perfetto/android"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/service/path"
+)
+
+const (
+	activityName   = "android.app.NativeActivity"
+	bufferSizeKb   = uint32(131072)
+	counterIdQuery = "" +
+		"select counter_id from counter_definitions " +
+		"where name = '%v'"
+	counterValuesQuery = "" +
+		"select value from counter_values " +
+		"where counter_id = %v order by ts " +
+		"limit %v offset 10"
+	counterPeriodNs                         = uint64(50000000)
+	durationMs                              = 7000
+	gpuCountersDataSourceDescriptorName     = "gpu.counters"
+	gpuRenderStagesDataSourceDescriptorName = "gpu.renderstages"
+	intentAction                            = "android.intent.action.MAIN"
+
+	// perfettoTraceFile is the location on the device where we'll ask Perfetto
+	// to store the trace data while tracing.
+	perfettoTraceFile     = "/data/misc/perfetto-traces/gapis-trace"
+	sampleCounter         = 100
+	validationPackageName = "com.samples.cube"
+)
+
+func Validate(ctx context.Context, p *path.Device) error {
+	if p == nil {
+		return log.Err(ctx, nil, "Invalid device path")
+	}
+	mgr := GetManager(ctx)
+	validator, ok := mgr.validators[p.ID.ID()]
+	if !ok || validator == nil {
+		return log.Errf(ctx, nil, "Could not find validator for device %d", p.ID.ID())
+	}
+	d := validator.GetDevice().(adb.Device)
+
+	// Get ActivityAction
+	packages, _ := d.InstalledPackages(ctx)
+	pkg := packages.FindByName(validationPackageName)
```

Comment:
As a baby step, I am hard coding this given that none of our code except `run gapit` will execute this path. The final shape of it would be integrating the cube as part of gapidapk, and hence this package name will eventually be the package name of gapidapk.